usr/sbin/p11sak/p11sak.c

```diff
@@ -2292,11 +2287,47 @@ static CK_RV generate_asymmetric_key(CK_SESSION_HANDLE session, CK_SLOT_ID slot,
         goto done;
     }
 
-    rc = set_battr(attr_string, pub_attr, &pub_acount, prv_attr, &prv_acount);
-    if (rc != CKR_OK) {
-        fprintf(stderr, "Error setting binary attributes (error code 0x%lX: %s)\n", rc,
-                p11_get_ckr(rc));
-        goto done;
+    /**
+     * if the separator sign is not in the string the custom attributes apply for 
+     * public and private key
+     * else the attributes in front of the separator sign are set for the public 
+     * key and the ones behind are set for the private key
+     */
+    char *ret = NULL;
+    if (attr_string) {
+        ret = strchr(attr_string, separator);
+    }
```

Comment:
No { ... } needed for a single line if-body.

Reply:
Done